audio/whisper/Portfile

```diff
@@ -0,0 +1,105 @@
+#-*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup   github  1.0
+
+name                whisper
+categories          audio
+github.setup        ggerganov whisper.cpp v1.6.2
+version             1.6.2
+platforms           darwin
+license             MIT
+maintainers         ijams.me:nate
+description         Port of OpenAI's Whisper model in C/C++
+long_description    High-performance inference of OpenAI's Whisper automatic speech recognition \
+                    (ASR) model
+notes               "
+Install this port's variants to include models, or download
+your own and use the built-in `-m` feature.
+
+Remember that whisper.cpp only functions on 16-bit WAV
+files. You can use `ffmpeg -i input.mp3 -ar 16000
+output.wav` to convert the file first.
+"
+homepage            https://github.com/ggerganov/whisper.cpp
+
+checksums           rmd160  4110b67e9b67405c4e28c1b92cd12471f3980731 \
+                    sha256  bc041f809c3476c4ebbb7eb1ffb2c447c8aaa85cd7b260c6426978dc53059810 \
+                    size    5089547
+
+build.target        main
+patchfiles          whisper-add-models.diff \
+                    ggml-hard-coded.diff \
+                    add-metal-path.diff
+
+post-patch {
+    reinplace "s|__PREFIX__|${prefix}|" \
+        whisper.cpp \
+        ggml-metal.metal \
+        ggml-metal.m
+}
+
+use_configure no
```

Comment:
It might make sense to use the `makefile` portgroup alongside the Github portgroup.  The `makefile` portgroup will pass a lot of platform-appropriate options into `make`, and it automatically defaults to `use_configure no`.